### PR TITLE
Add an action to generate metadata for dev deploys from `env.GITHUB_*`

### PR DESCRIPTION
**Pull request by remoun**
De-duplicate the `metadata.sh` helper across repos with dev deploys, including [`mobilecoin`](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/.internal-ci/util/metadata.sh), [`twix`](https://github.com/mobilecoinofficial/twix/blob/main/.internal-ci/util/metadata.sh), [`moby-handle-service`](https://github.com/mobilecoinofficial/moby-handle-service/blob/main/.internal-ci/util/metadata.sh)

Adapted (mostly copied) from https://github.com/mobilecoinfoundation/mobilecoin/blob/84b9d9250bb6af30b231f404d3be28d5a17928d9/.internal-ci/util/metadata.sh

### CODE REVIEW

File: util/metadata.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+# check for github reference variables.
+is_set GITHUB_REF_NAME
+is_set GITHUB_REF_TYPE
+is_set GITHUB_RUN_NUMBER
+is_set GITHUB_SHA
+is_set GITHUB_OUTPUT
+
+# Make sure prefix is less that 18 characters or k8s limits.
+namespace_prefix="mc"
```
[the remark this answers is not in the document]
Comment (jgreat) [reply util/metadata.sh:36]: maybe a default of of the repo name, something like `basename "${GITHUB_REPOSITORY}"` ?

File: util/metadata.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+if [[ -n "${DELETE_REF_NAME}" ]]
+then
+    branch="${DELETE_REF_NAME}"
+fi
+
+case "${GITHUB_REF_TYPE}" in
+    tag)
+        # check for valid tag and set outputs
+        version="${GITHUB_REF_NAME}"
+        if [[ ! "${version}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+.* ]]
```
Comment (jgreat) [inline util/metadata.sh:58]: Do we need to think about this pattern here and support the `v[0-9]+` pattern that twix uses.   I kinda wish we had just used the a bit more standard major.minor.patch, but there's a lot of tags now so :shrug:

**Review by jgreat — DISMISSED** [review]
This looks like its a copy of what's currently in mobilecoin, so works for me :grin:.  

With the namespace_prefix enhancements, will this work as is with other projects?  I know twix and others use just majors (`v[0-9]+`) as their versioning pattern.

File: util/metadata.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (remoun) [reply util/metadata.sh]: Added, as well as an optional `prefix` input to override

File: util/metadata.sh (hunk 2 above)
Comment (remoun) [reply util/metadata.sh]: Good catch! I made the minor and patch parts optional

**Review by jgreat — APPROVED** [review]
I like it!

**Pull request merged**